src/hera/volumes.py

```diff
@@ -115,6 +116,46 @@ def get_mount(self) -> V1VolumeMount:
         return V1VolumeMount(name=self.name, mount_path=self.mount_path)
 
 
+class SecretVolume(BaseModel):
+    """A volume representing a secret. This can be used to mount secrets to paths inside a task
+
+    Attributes
+    ----------
+    name: str
```

Comment:
My only concern here, is that we are taking some control away from the user. I could default then name to be the secret name but leave the name as optional. I'm thinking in an instance, that the secret name conflicts with an existing volume. Definitely an edge case, but could be possible.

Reply:
Ah, yes, that can certainly happen. As users can mount multiple volume types it's possible for the names to conflict. In light of this, I like your idea of making it optional. This is what happens, actually, with the other volumes:
```python
    name: Optional[str]

    @validator('name', always=True)
    def check_name(cls, value):
        """Validates that a name is specified. If not, it sets it"""
        if not value:
            return str(uuid.uuid4())
        return value
```
This can definitely be the case here as well. A UUID will hopefully not collide, but ultimately allows the user to specify the name as well.

Reply:
I think modifying the user's name selection is a bad idea (as discussed previously :-)).  It should be the user's responsibility to provide unique names. and the framework should not modify why the user provides, since that makes it very difficult if not impossible for the user to "find" it later.

Reply:
in this instance, we aren't modifying a user's input, rather supplying a dynamic default value. As an aside, an interesting way to do this could be via https://pydantic-docs.helpmanual.io/usage/models/#field-with-dynamic-default-value . Although that would go against defaulting to None.